satellite/metrics/counter.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package metrics
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc/rpcstatus"
+	"storj.io/storj/satellite/metainfo"
+)
+
+// counter implements the metainfo loop observer interface for data science metrics collection.
+//
+// architecture: Observer
+type counter struct {
+	remoteDependentObjectCount int64
+	inlineObjectCount          int64
+	totalObjectCount           int64
+}
+
+// newCounter instantiates a new counter to be subscribed to the metainfo loop.
+func newCounter() *counter {
+	return &counter{}
+}
+
+// Object increments counts for inline objects and remote dependent objects.
+func (counter *counter) Object(ctx context.Context, path metainfo.ScopedPath, pointer *pb.Pointer) (err error) {
+	streamMeta := &pb.StreamMeta{}
+	err = proto.Unmarshal(pointer.Metadata, streamMeta)
+	if err != nil {
+		return rpcstatus.Error(rpcstatus.Internal, err.Error())
+	}
+
+	if streamMeta.NumberOfSegments > 1 {
+		counter.remoteDependentObjectCount++
+	} else {
+		counter.inlineObjectCount++
+	}
```

Comment:
Maybe I'm missing something, but shouldn't this be:
```
	if len(pointer.GetRemote().GetRemotePieces()) >= 1 {
		counter.remoteDependentObjectCount++
	} else {
		counter.inlineObjectCount++
	}
```

Reply:
hm I think the `> 1` is okay, because the 1 segment should be the inline segment-- but I think I do need to add a secondary check for both cases to make sure that the respective pointer types are remote and inline. thanks!

Reply:
got it.  i was misreading the intent here.

Reply:
rewrote so it should be less confusing either way